src/Compilers/CSharp/Portable/Symbols/TupleTypeSymbol.cs

```diff
@@ -692,29 +779,55 @@ internal override ImmutableArray<NamedTypeSymbol> GetInterfacesToEmit()
     internal sealed class TupleFieldSymbol : FieldSymbol
     {
         private readonly string _name;
-        private readonly FieldSymbol _underlyingFieldOpt;
         private readonly TypeSymbol _type;
         private readonly TupleTypeSymbol _containingTuple;
+        private readonly int _index;
+        private readonly FieldSymbol _underlyingField;
 
         /// <summary>
         /// Missing underlying field is handled for error recovery
         /// A tuple without backing fields is usable for binding purposes, since we know its name and type,
         /// but caller is supposed to report some kind of error at declaration.
+        ///
+        /// index is 0 for Item1.
         /// </summary>
-        public TupleFieldSymbol(string name, TupleTypeSymbol containingTuple, TypeSymbol type, FieldSymbol underlyingFieldOpt)
+        internal TupleFieldSymbol(string name, TupleTypeSymbol containingTuple, TypeSymbol type, int index, FieldSymbol underlyingField)
         {
             _name = name;
             _containingTuple = containingTuple;
+            _underlyingField = underlyingField;
             _type = type;
-            _underlyingFieldOpt = underlyingFieldOpt;
+            _index = index;
         }
 
-        public FieldSymbol UnderlyingTypleFieldSymbol
+        /// <summary>
+        /// Copy this tuple field, but modify it to use the new containing tuple and underlying type.
+        /// </summary>
+        internal TupleFieldSymbol WithUnderlyingType(TupleTypeSymbol newContainingTuple, NamedTypeSymbol newContainingUnderlyingType)
         {
-            get
+            FieldSymbol newUnderlyingField = _underlyingField.OriginalDefinition.AsMember(newContainingUnderlyingType);
+            return new TupleFieldSymbol(_name, newContainingTuple, newUnderlyingField.Type, _index, newUnderlyingField);
+        }
+
+        internal static FieldSymbol GetUnderlyingField(int containingTupleArity, NamedTypeSymbol containingUnderlyingType, int fieldIndex, CSharpSyntaxNode syntax, Binder binder, DiagnosticBag diagnostics)
+        {
+            int tupleRemainder;
+            int tupleChainLength = TupleTypeSymbol.NumberOfValueTuples(containingTupleArity, out tupleRemainder);
+
+            int fieldRemainder;
+            int fieldChainLength = TupleTypeSymbol.NumberOfValueTuples(fieldIndex + 1, out fieldRemainder);
+
+            int containingLinkSize = 8;
+            if (fieldChainLength == tupleChainLength)
             {
-                return _underlyingFieldOpt;
+                containingLinkSize = tupleRemainder;
             }
+
+            WellKnownMember wellKnownTypeMember = TupleTypeSymbol.GetTupleTypeMember(containingLinkSize, fieldRemainder);
+            var linkField = (FieldSymbol)Binder.GetWellKnownTypeMember(binder.Compilation, wellKnownTypeMember, diagnostics, syntax: syntax);
+            NamedTypeSymbol linkType = TupleTypeSymbol.GetNestedTupleType(containingUnderlyingType, fieldChainLength - 1);
```

Comment:
Fixed.
But I will keep the general error handling change for symbol construction to a separate commit.